include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -162,6 +162,12 @@ class IExecutionProvider {
   */
   virtual common::Status OnRunEnd(bool /*sync_stream*/) { return Status::OK(); }
 
+  virtual void CaptureBegin() {}
```

Comment:
Just a thought - Would moving CaptureBegin() and CaptureEnd() into OnRunStart() and OnRunEnd() respectively be a better idea ? If the CUDA EP is configured to use cuda graphs, we could conditionally invoke CaptureBegin() and CaptureEnd()... Please see https://github.com/microsoft/onnxruntime/pull/9978/files#r795489852